src/Layer.js

```diff
@@ -371,84 +383,73 @@ export default class Layer {
      * Custom Layer API: `prerender` function
      */
     prerender (gl, matrix) {
-        // Call request data if the matrix has changed
-        if (!util.equalArrays(this._matrix, matrix)) {
-            this._matrix = matrix;
-            this.renderer.matrix = matrix;
-            this._setRendererZoomCenter(matrix);
-            if (this._source && this._visible) {
-                this._source.requestData(this._getZoom(), this._getViewport());
-            }
-            this._fireUpdateOnNextRender = true;
-        } else if (!this._noFirstRequestData && this._source && this._visible) {
-            this._source.requestData(this._getZoom(), this._getViewport());
-            this._noFirstRequestData = true;
+        this.renderer.matrix = matrix;
+        if (this._source && this.visible) {
+            this._source.requestData(this._getZoom(), this._getViewport(matrix));
         }
     }
```

Comment:
There are at least 2 reasons (this was simplified step by step).

One reason is dead code like `_setRendererZoomCenter` which did things, but those things were never used.

The other reason is that before there was a lot of effort into trying to minimize calls to `requestData`. However, I think that wasn't good because it added a lot of code and the `requestData` function should be optimized (and it is) anyway because of a simple thing like panning will modify the view matrix at every frame during the panning